python/PythonPathFinderPass.cpp

```diff
@@ -0,0 +1,125 @@
+//===- PathfinderFlowsWithPython.cpp ----------------------------*- C++ -*-===//
+//
+// Copyright (C) 2022, Advanced Micro Devices, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PybindTypes.h"
+
+#include "aie/Dialect/AIE/IR/AIEDialect.h"
+#include "aie/Dialect/AIE/Transforms/AIEPasses.h"
+#include "aie/Dialect/AIE/Transforms/AIEPathFinder.h"
+
+#include "mlir/Bindings/Python/PybindAdaptors.h"
+#include "mlir/CAPI/Pass.h"
+#include "mlir/IR/BuiltinDialect.h"
+#include "mlir/Pass/Pass.h"
+
+using namespace mlir;
+using namespace mlir::python::adaptors;
+using namespace xilinx::AIE;
+
+namespace py = pybind11;
+
+//    ┌─────┐   ┌─────┐
+//    │ 1,0 ├   ┤ 1,1 │
+//    │     │ N │     │
+//    └─────┘   └─────┘
+//       W         E
+//    ┌─────┐   ┌─────┐
+//    │ 0,0 │ S │ 0,1 │
+//    │     │   │     │
+//    └─────┘   └─────┘
+
+class PythonPathFinder : public Pathfinder {
```

Comment:
"Pathfinder" is a particular routing algorithm.  I think if you're solving an ILP, you're probably not doing the pathfinder algorithm?

Reply:
https://ieeexplore.ieee.org/document/1377269